core/objects.go

```diff
@@ -2,12 +2,36 @@ package core
 
 import "fmt"
 
-// tile defines how a tile is handled by the super-nes
-type tile struct {
-	number       uint16 // tile number (used to find it in the vram)
-	palette      uint8  // index of the color palette to use (the number of entries in the palette depends on the mode and the background)
-	priority     bool   // Tile priority, for tiles this is only encoded in 1 bit (whereas for sprites it's encoded into 2 bits)
-	hFlip, vFlip bool   // horizontal and vertical flips
+//baseTile represents an 8x8 pixels tile
+type baseTile struct {
+	addr       uint16
+	palette    uint8
+	colorDepth uint8
+}
+
+//baseTileSize returns the size of a base tile in bytes depending on its color depth
+func baseTileSize(colorDepth uint8) uint16 {
+	//base tile size in bits = number of pixels * number of bits per pixel = 64 * colorDepth
+	//base tile size in bytes = base tile size in bits / 8 = 8 * colorDepth
+	return 8 * uint16(colorDepth)
+}
+
+// bgTile represents a variable size background tile
+type bgTile struct {
+	palette       uint8  // index of the color palette to use (the number of entries in the palette depends on the mode and the background)
```

Comment:
nit: Should we embed a field of type `baseTile` here to avoid repetition with the `baseTile` attributes (palette, addr and colorDepth)